src/composables/useSegmentGroupConfigInitializer.ts

```diff
@@ -0,0 +1,21 @@
+import useLayerColoringStore from '@/src/store/view-configs/layers';
+import { watchImmediate } from '@vueuse/core';
+import { MaybeRef, computed, unref } from 'vue';
+
+export function useSegmentGroupConfigInitializer(
+  viewId: MaybeRef<string>,
+  layerId: MaybeRef<string>
+) {
+  const coloringStore = useLayerColoringStore();
+  const colorConfig = computed(() =>
+    coloringStore.getConfig(unref(viewId), unref(layerId))
+  );
+
+  watchImmediate(colorConfig, (config) => {
+    if (config) return;
+
+    const viewIdVal = unref(viewId);
+    const layerIdVal = unref(layerId);
+    coloringStore.updateConfig(viewIdVal, layerIdVal, {});
```

Comment:
Is there a reason we need an initializer composable that initializes the view/layer key to an empty object?

Reply:
The purpose of the initializer is to create the config entry for the segment group view in the coloring store.   

Passing the empty object got merged with the default config values.  Does read strange.  Added a coloringStore.initConfig() function.